src/event_data.c

```diff
@@ -164,13 +164,12 @@ bool32 CanResetRTC(void)
 
 u16 *GetVarPointer(u16 id)
 {
-    if (id < 0x4000)
+    if (id < VARS_START)
         return NULL;
-
-    if ((s16)id >= 0)
-        return &gSaveBlock1Ptr->vars[id - 0x4000];
-
-    return gSpecialVars[id - 0x8000];
+    else if (id < VAR_0x8000)
```

Comment:
I'd use `SPECIAL_VARS_START` for 0x8000, instead of just treating it as a regular variable. 0x4000-0x7FFF variables are conceptually different from 0x8000-0xFFFF ones, so I would split the enums.